src/providers/openstack/configdrive.rs

```diff
@@ -0,0 +1,166 @@
+//! configdrive metadata fetcher for OpenStack
+//! reference: https://docs.openstack.org/nova/latest/user/metadata.html
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+
+use openssh_keys::PublicKey;
+use slog_scope::{error, warn};
+use tempfile::TempDir;
+
+use crate::errors::*;
+use crate::network;
+use crate::providers::MetadataProvider;
+
+const CONFIG_DRIVE_LABEL: &str = "config-2";
+
+/// OpenStack config-drive.
+#[derive(Debug)]
+pub struct OpenstackConfigDrive {
+    /// Path to the top directory of the mounted config-drive.
+    drive_path: PathBuf,
+    /// Temporary directory for own mountpoint (if any).
+    temp_dir: Option<TempDir>,
+}
+
+impl OpenstackConfigDrive {
+    /// Try to build a new provider client.
+    ///
+    /// This internally tries to mount (and own) the config-drive.
+    pub fn try_new() -> Result<Self> {
+        // Short-circuit if the config-drive is already mounted.
+        let path = Path::new("/media/ConfigDrive/openstack/latest/");
+        if path.exists() {
+            return Ok(OpenstackConfigDrive {
+                temp_dir: None,
+                drive_path: PathBuf::from("/media/ConfigDrive/"),
+            });
+        }
+
+        // Otherwise, try and mount with the label.
+        let target = tempfile::Builder::new()
+            .prefix("afterburn-")
+            .tempdir()
+            .chain_err(|| "failed to create temporary directory")?;
+        crate::util::mount_ro(
+            &Path::new("/dev/disk/by-label/").join(CONFIG_DRIVE_LABEL),
+            target.path(),
+            "iso9660",
```

Comment:
Let's split out this constant to its own declaration.

Reply:
Added `const TARGET_FS: &str = "iso9660";` for the file system type.